src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -390,12 +391,16 @@ public boolean isRunning() {
 	@Override
 	protected Object handleRequestMessage(final Message<?> message) {
 		final ProducerRecord<K, V> producerRecord;
+		boolean flush = Boolean.TRUE.equals(message.getHeaders().get(KafkaIntegrationHeaders.FLUSH));
```

Comment:
May it be more broad and flexible? Kinda `flushExpression`, like many other per-message options?